src/main/java/com/github/qwazer/markdown/confluence/core/service/MainService.java

```diff
@@ -20,30 +25,89 @@
     private final FileReaderService fileReaderService;
     private MarkdownService markdownService;
     private final PageService pageService;
+    private final AttachmentService attachmentService;
+    private final ConfluenceService confluenceService;
 
     @Autowired
     public MainService(FileReaderService fileReaderService,
                        MarkdownService markdownService,
-                       PageService pageService) {
+                       PageService pageService,
+                       AttachmentService attachmentService,
+                       ConfluenceService confluenceService) {
         this.fileReaderService = fileReaderService;
         this.markdownService = markdownService;
         this.pageService = pageService;
+        this.attachmentService = attachmentService;
+        this.confluenceService = confluenceService;
     }
 
     public void processAll(ConfluenceConfig confluenceConfig) throws IOException {
 
+        confluenceService.setConfluenceConfig(confluenceConfig);
+
         if (confluenceConfig.getParseTimeout() != DEFAULT_PARSE_TIMEOUT) {
             this.markdownService = new MarkdownService(confluenceConfig.getParseTimeout());
         }
 
         List<Page> orderedList = order(confluenceConfig.getPages());
         for (Page page : orderedList) {
-            String plainFileContent = fileReaderService.readFile(page);
-            String wikiText = markdownService.convertMarkdown2Wiki(plainFileContent, confluenceConfig);
-            pageService.postWikiPageToConfluence(page, confluenceConfig, wikiText);
+            String markdownText = fileReaderService.readFile(page);
+            String wikiText = markdownService.convertMarkdown2Wiki(markdownText, confluenceConfig);
+
+            // Holds the original uri and, if its a file uri, the path to the file
+            Map<String, Path> imageURIToFilePathMap = new HashMap<>();
+            wikiText = processInlineImages(markdownText, wikiText, page.getSrcFile(), imageURIToFilePathMap);
+
+            Long pageId = pageService.postWikiPageToConfluence(page, wikiText);
+
+            for (Map.Entry<String, Path> entry : imageURIToFilePathMap.entrySet()) {
+                attachmentService.postAttachmentToPage(pageId, entry.getValue());
+            }
         }
     }
 
+    // Any images in the markdown need to be published as attachments to the page.
```

Comment:
this feature and how to use 
should be part of plugin doc  (README.md)

Reply:
Done